### DISCUSSION TITLE
How to uninstall Barrier on Kubuntu?

### DISCUSSION THREAD

**Discussion by AvgOrdinaryGuy** [debauchee/barrier#1276] [Q&A]
Installed Barrier complied from source hoping to correct issues that developed on my snap install.  After uninstalling the snap package, I installed from source, but still have the same issues.  How can a source install be uninstalled please?

**Comment by shymega** [comment]
If you used `make`, it's not possible as our build system - CMake - does not enable an uninstall target by default. 

I do not believe we have plans for an uninstall target at this time.

If you look at the RPM spec, you'll see a list of installed files, and you can delete them from there. You will need root perms.

Below is a command to delete all listed files the RPM spec declares. Use with caution, I take **no** responsibility if this messes up your system - you, and you alone should use your discretion whenever to run the command.

It is derived from the RPM spec and only deletes from the `/usr/local` prefix, which the install target would normally install. If you used a different prefix, I can't help you there.

```
sudo rm -v /usr/local/bin/barrier /usr/local/bin/barrierc /usr/local/bin/barriers /usr/local/man/man1/barrierc.1 /usr/local/man/man1/barriers.1 /usr/local/share/applications/barrier.desktop /usr/local/share/icons/hicolor/scalable/apps/barrier.svg
```

**Comment by AvgOrdinaryGuy** [comment]
Thanks shymega, that got the job done.  However, some configuration files must still remain, because after removing and re-installing from repo, there was no new install setup routine offered, and Barrier is trying to run my old configurations that don't exist any longer.  That leads for failure to start.

**Comment by shymega** [comment]
Look at the .config directory in your HOME.

On Sat, Jul 10, 2021, at 8:05 PM, AvgOrdinaryGuy wrote:
> 
> 
> Thanks shymega, that got the job done. However, some configuration 
> files must still remain, because after removing and re-installing from 
> repo, there was no new install setup routine offered, and Barrier is 
> trying to run my old configurations that don't exist any longer. That 
> leads for failure to start.
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/debauchee/barrier/issues/1222#issuecomment-877688259>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAKF2QBBXUYSVB64PL4IQ5DTXCKXXANCNFSM47YOKX4A>.
> 


-- 
Kind regards,
--
Dom Rodriguez (also known as shymega)

**Comment by AvgOrdinaryGuy** [comment]
Thanks again.  Found a barrier config file in .config/Debauchee/ that I had overlooked before.

**Comment by shymega** [comment]
Good, closing now.
